### PR TITLE
Added option to override the filename when creating attachments

### LINKED ISSUE #141: Add option to override the name of an attached file
It would be useful to be able to set the name of a file that is uploaded manually instead of relying on the path.

We have a form that people use to create issues in JIRA and to implement the file upload we have to move the file to a temp location with it's real name before submitting to JIRA. I'd like to be able to do this without having to move the file.

I can do a PR for this but wanted to get an opinion first.

### PR DESCRIPTION
**Pull request by betterphp**
Closes #141

### CODE REVIEW

File: src/Jira/Api/Client/CurlClient.php [hunk 1]
```diff
@@ -159,16 +159,21 @@ public function sendRequest(
 	 * If necessary, replace curl file @ string with a CURLFile object (for PHP 5.5 and up)
 	 *
 	 * @param string $file_string The string in @-format as it is used on PHP 5.4 and older.
+	 * @param string $name Optional override for the filename
 	 *
 	 * @return \CURLFile|string
 	 */
-	protected function getCurlValue($file_string)
+	protected function getCurlValue($file_string, $name)
```
Comment (aik099) [inline src/Jira/Api/Client/CurlClient.php]: Please change `$name` into `$name = null`.

File: src/Jira/Api/Client/CurlClient.php [hunk 2]
```diff
@@ -159,16 +159,21 @@ public function sendRequest(
...
 	{
+		if ($name === null){
```
Comment (aik099) [inline src/Jira/Api/Client/CurlClient.php]: CS: incorrect spacing.

File: src/Jira/Api.php [hunk 3]
```diff
@@ -644,17 +644,15 @@ public function releaseVersion($version_id, $release_date = null, array $params
 	 *
 	 * @param string $issue_key Issue key.
 	 * @param string $filename  Filename.
-	 * @param array  $options   Options.
+	 * @param string $name      Name.
```
Comment (aik099) [inline src/Jira/Api.php]: CS: Probably incorrectly aligned DocBlock.

Comment (betterphp) [inline src/Jira/Api.php]: I can't really see what's wrong here :/

**Comment by betterphp** [comment]
When I run phpcs it reports about 40 errors so I assumed it was not really used. Could that be because of my local PHP version?

**Comment by betterphp** [comment]
Also sorry for the delay, I'm doing this for work and have been moved on to another project for a while

File: src/Jira/Api.php (hunk 3 above)
Comment (aik099) [inline src/Jira/Api.php]: Maybe it's just GitHub replaced all TABs with SPACEs and it look wrong 😉

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,6 +15,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 - Added `Api::getProjectIssueTypes` call for getting project issue types and issue statuses connected to them by [@N-M].
 - Added `Api::getResolutions` call for getting available issue resolutions by [@N-M].
 - Allow configuring issues queried per page in `Walker` class by [@aik099].
+- Added optional override for the filename in Api::createAttachment by [@betterphp]
```
Comment (aik099) [inline CHANGELOG.md]: Not done.

**Comment by aik099** [comment]
I've run PHP_CodeSniffer only inspection on files changed in PR and it reported 7 issues and none were in the changed lines. It's also important to use correding PHPCS standard (see contribution instructions) and not default PEAR one.

**Comment by aik099** [comment]
Merging, thanks @betterphp .

**Pull request merged**